pkg/model/keycloak_deployment.go

```diff
@@ -349,3 +351,29 @@ func KeycloakVolumes() []v1.Volume {
 		},
 	}
 }
+
+// We allow the patch version of an image for keycloak to be modified outside of the operator on the cluster
+func getReconciledKeycloakImage(currentState *v13.StatefulSet) string {
+	currentImage := GetCurrentKeycloakImage(currentState)
+	currentImageRepo, currentImageMajor, currentImageMinor := getKeycloakImageRepoAndVersion(currentImage)
+	keycloakImageRepo, keycloakImageMajor, keycloakImageMinor := getKeycloakImageRepoAndVersion(KeycloakImage)
+
+	// Check the repos, major and minor versions match. If so, return and reconcile with the current cluster image
+	// E.g. quay.io/keycloak/keycloak:7.0.1
+	if currentImageRepo == keycloakImageRepo && currentImageMajor == keycloakImageMajor && currentImageMinor == keycloakImageMinor {
+		return currentImage
+	}
+
+	return KeycloakImage
```

Comment:
I just realized that this breaks automatic z-stream migrations!

Let's consider the following scenario, we have a deployed Keycloak with version `7.0.0` (`currentImage`) and we just installed a newer version of the operator, that uses `7.0.1` (`KeycloakImage`). In this case, both major and minor component match. However, the image won't be upgraded!

Reply:
excellent spotting @slaskawi . Fixed here https://github.com/keycloak/keycloak-operator/pull/70